Build failure with default_features = false

### Observed behavior

When depending on `async-nats = "0.34.0`, if I set `default-features = false`, it fails to build with the following error:

```
error[E0560]: struct `jetstream::stream::Config` has no field named `compression`
    --> /home/jaskij/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-nats-0.34.0/src/jetstream/context.rs:1048:17
     |
1048 |                 compression: if config.compression {
     |                 ^^^^^^^^^^^ `jetstream::stream::Config` does not have this field
     |
     = note: available fields are: `max_bytes`, `max_messages`, `max_messages_per_subject`, `discard_new_per_subject`, `retention` ... and 12 others

For more information about this error, try `rustc --explain E0560`.
error: could not compile `async-nats` (lib) due to 1 previous error
```

### Expected behavior

The crate builds and works as intended.

### Server and client version

from my Cargo.toml
```toml
async-nats = { version = "0.34.0", default-features = false }
```

### Host environment

up-to-date Arch Linux running on x86-64, building with current Rust stable

```
$ cargo --version
cargo 1.77.1 (e52e36006 2024-03-26)
```

### Steps to reproduce

Make a Rust crate depending on `async-nats` 0.34.0 with `default-features = false`. It fails to build.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -123,6 +123,26 @@ jobs:
       - name: Check lint
         run: cargo doc --no-deps --all-features
 
+  check_features:
+    name: check feature flags (ubuntu-latest / stable)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+      - name: Install stable Rust on ubuntu-latest
+        id:   install-rust
+        uses: dtolnay/rust-toolchain@stable
+      - name: Restore cached build artifacts
+        uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: ${{ runner.os }}-${{ steps.install-rust.outputs.cachekey }}-${{ hashFiles('**/Cargo.toml') }}
+          save-if: false # we only build docs, so we don't have the full build to be cached.
+      - name: Install cargo hack
+        run: cargo install cargo-hack
+      - name: Check lint
+        run: cargo hack check --feature-powerset --no-dev-deps -p async-nats
+
+
   check_licenses:
     name: check licenses (ubuntu-latest / stable)
     runs-on: ubuntu-latest

--- a/async-nats/src/jetstream/context.rs
+++ b/async-nats/src/jetstream/context.rs
@@ -41,9 +41,11 @@ use super::errors::ErrorCode;
 use super::is_valid_name;
 use super::kv::{Store, MAX_HISTORY};
 use super::object_store::{is_valid_bucket_name, ObjectStore};
+#[cfg(feature = "server_2_10")]
+use super::stream::Compression;
 use super::stream::{
-    self, Compression, Config, ConsumerError, ConsumerErrorKind, DeleteStatus, DiscardPolicy,
-    External, Info, Stream,
+    self, Config, ConsumerError, ConsumerErrorKind, DeleteStatus, DiscardPolicy, External, Info,
+    Stream,
 };
 
 /// A context which can perform jetstream scoped requests.
@@ -1045,6 +1047,7 @@ impl Context {
                 discard: DiscardPolicy::New,
                 allow_rollup: true,
                 allow_direct: true,
+                #[cfg(feature = "server_2_10")]
                 compression: if config.compression {
                     Some(Compression::S2)
                 } else {

--- a/async-nats/tests/client_tests.rs
+++ b/async-nats/tests/client_tests.rs
@@ -28,7 +28,6 @@ mod client {
     async fn force_reconnect() {
         let (dctx, mut dcrx) = tokio::sync::mpsc::channel(1);
         let (rctx, mut rcrx) = tokio::sync::mpsc::channel(1);
-        let (tx, mut rx) = tokio::sync::mpsc::channel(1);
 
         let server = nats_server::run_basic_server();
 
@@ -47,16 +46,6 @@ mod client {
             .connect(server.client_url())
             .await
             .unwrap();
-
-        let second_client = async_nats::connect(server.client_url()).await.unwrap();
-        // second client to check if we properly flush before forcing reconnect.
-        let mut sub = second_client.subscribe("test").await.unwrap();
-        tokio::task::spawn({
-            async move {
-                sub.next().await.unwrap();
-                tx.send(()).await.unwrap();
-            }
-        });
 
         let mut sub = client.subscribe("foo").await.unwrap();
 
@@ -78,12 +67,6 @@ mod client {
         client.publish("foo", "data".into()).await.unwrap();
 
         tokio::time::timeout(Duration::from_secs(5), sub.next())
-            .await
-            .unwrap()
-            .unwrap();
-
-        // check if message sent just before reconnect is received.
-        tokio::time::timeout(Duration::from_secs(15), rx.recv())
             .await
             .unwrap()
             .unwrap();

